.pre-commit-config.yaml

```diff
@@ -165,7 +165,7 @@ repos:
   hooks:
   - id: pytest
     name: Run unit tests
-    entry: pytest -c pyproject.toml --cov-config pyproject.toml src/package/ tests/ docs/
+    entry: env COVERAGE_CORE=sysmon pytest -c pyproject.toml --cov-config pyproject.toml src/package/ tests/ docs/
```

Comment:
Please add a comment based on the PR description to explain why `COVERAGE_CORE=sysmon` is necessary here, and also mention that this option is only recommended for Python version > 3.12.